### ISSUE TITLE
fix(parser-utils): exclude "/" from unquoted attribute value terminators (and fix offset corruption)

### ISSUE THREAD

**Issue by YusukeHirao** [markuplint/markuplint#3594]
## Describe the bug

The HTML parser fails to parse unquoted attribute values containing `/`, such as `type=image/gif`. This is valid HTML — the HTML spec allows unquoted attribute values as long as they don't contain specific characters (space, `"`, `'\''`, `=`, `<`, `>`、`` ` ``), and `/` is permitted.

> https://html.spec.whatwg.org/multipage/syntax.html#attributes-2

## Code Example

```html
<picture>
  <source srcset=x type=image/gif>
  <img src=x alt=x>
</picture>
```

## The current behavior

```
SyntaxError: Invalid tag syntax: "<source srcset=x type=image/gif>"
```

## The expected behavior

No parse error. The parser should accept `/` in unquoted attribute values.

## Impact

Causes 2 parse errors (false positives) in the nu-html-checker compatibility benchmark.

## Affected package

`@markuplint/html-parser`

**Comment by YusukeHirao** [comment]
### Investigation result

**Root cause**: `@markuplint/parser-utils`'s `attrTokenizer` includes `/` in `endOfUnquotedValueChars` by default (line 26 of `attr-tokenizer.ts`):

```typescript
endOfUnquotedValueChars: ReadonlyArray<string> = [...defaultSpaces, '/', '>'],
```

This causes `type=image/gif` to be parsed as `type=image` (value ends at `/`), then `/gif>` triggers a parse error.

**HTML spec says**: Unquoted attribute values must not contain `"`, `'`, `=`, `<`, `>`, `` ` ``, or whitespace. **`/` is not in this list** — it's valid in unquoted values.

However, `parser-utils` treats `/` as an end-of-value char because it doubles as a self-closing solidus (`<br/>`). The issue is that `image/gif>` and `<br/>` are ambiguous without lookahead.

**Fix approach**: For the HTML parser specifically, the `endOfUnquotedValueChars` could be overridden. However, `html-parser` delegates attribute parsing to `parser-utils`'s `Parser` base class, and the `visitAttr` call happens internally with no hook to customize `endOfUnquotedValueChars`.

**Possible solutions**:
1. Change the default to not include `/` — may break template parsers (pug already overrides to `[]`)
2. Add a parser option to configure `endOfUnquotedValueChars` at the `Parser` class level
3. Use lookahead: if `/` is followed by `>`, treat as self-closing; otherwise, include in value

This needs further design work. Option 3 is the most correct but most complex.

**Comment by YusukeHirao** [comment]
## nu-html-checker 実出力での検証

Docker 経由で nu-html-checker を実行して確認:

### テスト HTML
```html
<picture><source srcset=x type=image/gif><img src=x alt=x></picture>
```

### nu-html-checker 出力
**0 error** — クリーンにパース

### markuplint 出力
```
SyntaxError: Invalid tag syntax: '<source srcset=x type=image/gif>'
```

**さらに**: この parse error 発生後、後続のエラーレポートで line offset が崩れる（別件の副次バグ）。

### 判定: CONFIRMED（parser バグ + 副次的レポート破壊）

HTML 仕様上 `/` は unquoted attribute value 内で許容される。markuplint の html-parser の実装ミス。

### 関連箇所
`@markuplint/html-parser` の tag token 解析処理。nu-validator ベンチマークで 2 件の parse error を発生させている。

**Comment by YusukeHirao** [comment]
## ⚠️ 過去コメントは未検証

本 Issue の 2026-04-14 以前のコメントは nu-html-checker を実行せずに書かれた推測ベースです。以下が Docker 実測結果です。

## 実測結果（2026-04-15）

### 全 3 ケース対比

| # | 入力 | nu-html-checker | markuplint | Gap |
|---|------|----------------|-----------|-----|
| A | `<picture><source srcset=x type=image/gif><img src=x alt=x></picture>` | 違反なし（正常パース） | `SyntaxError: Invalid tag syntax: "<source srcset=x type=image/gif>"` (`parse-error`) | **YES**（偽のパースエラー） |
| B | `<script src=/foo.js></script>` | 違反なし | `SyntaxError: Invalid tag syntax: "<script src=/foo.js>"` | **YES** |
| C | `<img src=/a/b alt=x>` | 違反なし | `SyntaxError: Invalid tag syntax: "<img src=/a/b alt=x>"` | **YES** |

## 副次バグ: レポート破壊

parse error 発生後、後続行の line/column オフセットが崩れる。3 ケース全てで確認。error report 全体の信頼性に影響。

## 真のスコープ

- HTML 仕様上、unquoted attribute value 内で `/` は許可されている（禁止文字は ` `, `"`, `'`, `=`, `<`, `>`, `` ` `` のみ）
- 現行の `@markuplint/parser-utils` の `attrTokenizer` は `/` を `endOfUnquotedValueChars` として扱っているため誤パース
- 副次的にエラーレポートのオフセット計算も破壊している

## 実装方針

`@markuplint/parser-utils` の `attrTokenizer`:
- `endOfUnquotedValueChars` から `/` を除外
- `/>` 連続を自己閉じタグの solidus として特別扱い（lookahead）

副次バグも同時修正推奨。
